src/controller/cve.controller/cve.controller.js

```diff
@@ -302,10 +302,83 @@ async function submitCna (req, res, next) {
   }
 }
 
+async function updateCna (req, res, next) {
+  try {
+    const id = req.ctx.params.id
+    const cveRepo = req.ctx.repositories.getCveRepository()
+    const cveIdRepo = req.ctx.repositories.getCveIdRepository()
+    const orgRepo = req.ctx.repositories.getOrgRepository()
+    const userRepo = req.ctx.repositories.getUserRepository()
+    const orgUuid = await orgRepo.getOrgUUID(req.ctx.org)
+    const userUuid = await userRepo.getUserUUID(req.ctx.user, orgUuid)
+
+    // check that cve id exists
+    let result = await cveIdRepo.findOneByCveId(id)
+    if (!result || result.state === CONSTANTS.CVE_STATES.AVAILABLE) {
+      return res.status(400).json(error.cveDne())
+    }
+
+    // check that cveId org matches user org
+    const cveId = result
+    const isSecretariat = await orgRepo.isSecretariat(req.ctx.org)
+    if ((cveId.owning_cna !== orgUuid) && !isSecretariat) {
+      return res.status(403).json(error.owningOrgDoesNotMatch())
+    }
+
+    // check that cve record does exist
+    result = await cveRepo.findOneByCveId(id)
+    if (!result) {
+      return res.status(403).json(error.cveRecordDne())
+    }
+
+    // update cve record here
+    const cveRecord = result.cve
+    const cnaContainer = req.body.cnaContainer
+    cveRecord.cveMetadata.updated = (new Date()).toISOString()
+    if (cveRecord.cveMetadata.state !== CONSTANTS.CVE_STATES.PUBLISHED) {
+      cveRecord.cveMetadata.state = CONSTANTS.CVE_STATES.PUBLISHED
+    }
+    const providerMetadata = Object.assign({}, cveRecord.containers.cna.providerMetadata)
+    providerMetadata.updated = (new Date()).toISOString()
+    cnaContainer.providerMetadata = providerMetadata
+    cveRecord.containers.cna = cnaContainer
+    const cveModel = new Cve({ cve: cveRecord })
+    result = Cve.validateCveRecord(cveModel.cve)
+    if (!result) {
+      return res.status(500).json(error.serverError())
+    }
+
+    // change cve id state to publish
+    if (cveId.state === CONSTANTS.CVE_STATES.REJECTED) {
+      await cveIdRepo.updateByCveId(id, { state: CONSTANTS.CVE_STATES.PUBLISHED })
```

Comment:
Make sure to check the error and if there is one, fire off an internal server error message. we can't let the ID get out of sync with the record